src/pages/repository/components/tabs/settings-tab/General.tsx

```diff
@@ -32,29 +41,49 @@ export default function General() {
   const {
     register: registerTitle,
     handleSubmit: handleTitleSubmit,
-    formState: { errors: titleErrors }
+    formState: { errors: titleErrors, isValid: isTitleValid, isDirty: isTitleDirty }
   } = useForm({
-    resolver: yupResolver(titleSchema)
+    resolver: yupResolver(titleSchema),
+    mode: 'onChange'
   })
   const {
     register: registerDescription,
     handleSubmit: handleDescriptionSubmit,
-    formState: { errors: descriptionErrors }
+    formState: { errors: descriptionErrors, isValid: isDescriptionValid, isDirty: isDescriptionDirty }
   } = useForm({
     resolver: yupResolver(descriptionSchema)
   })
 
   async function handleUpdateButtonClick(data: yup.InferType<typeof descriptionSchema>) {
-    if (selectedRepo) {
+    if (selectedRepo && selectedRepo.description !== data.description) {
+      setIsSubmittingDescription(true)
       await updateDescription(data.description)
       toast.success('Successfully updated repository description')
+      setIsSubmittingDescription(false)
     }
   }
 
   async function handleRenameButtonClick(data: yup.InferType<typeof titleSchema>) {
-    if (selectedRepo) {
-      await updateName(data.title)
-      toast.success('Successfully updated repository name')
+    if (selectedRepo && data.title !== selectedRepo.name) {
```

Comment:
can we add a else condition with an error toast message saying "The new name you have entered is the same as your current name."